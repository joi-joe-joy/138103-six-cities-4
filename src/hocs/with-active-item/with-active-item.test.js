import React from 'react';
import renderer from "react-test-renderer";
import withActiveItem from "./with-active-item";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

const MockComponent = () => <div/>;

const MockComponentWrap = withActiveItem(MockComponent);

const offers = [
  {
    title: `Canal View Prinsengracht`,
    premium: true,
    pictures: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/studio-photos.jpg`
    ],
    price: 280,
    rating: 3.8,
    type: `house`,
    bedrooms: 3,
    amenities: [`Indoor fireplace`, `Kitchen`, `Wifi`, `Washer`],
    maxGuestsNumber: `Max 2 adults, 1 children`,
    description: `The apartment has a Queen size bed and a very wide modern couch that turns into a bed for a single person by moving the single portion of the couch to the bottom part of the love seat portion of the couch to create a long comfortable bed for a 3rd person. I hope you will enjoy the decor, it has a european feel towards the French side. You will be in the heart of the East Village in Manhattan, considered to be one of the most exciting and dynamic neighborhoods of the world.`,
    host: {
      photo: ``,
      name: `John Donn`,
      super: true
    },
    coordinations: [52.3809553943508, 4.939309666406198]
  }
];

const mockStore = configureStore([]);

it(`should render withActiveItem correctly`, () => {
  const store = mockStore({
    hoverCard: null
  });
  const tree = renderer.create(
      <Provider store={store}>
        <MockComponentWrap
          type={`cities`}
          offers={offers}
        />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
