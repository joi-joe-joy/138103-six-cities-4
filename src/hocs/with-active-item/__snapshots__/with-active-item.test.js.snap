// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render withActiveItem correctly 1`] = `<div />`;
